### DISCUSSION TITLE
RAG and Open WebUI very basic question

### DISCUSSION THREAD

**Discussion by Greg-Obleshchuk** [open-webui/open-webui#7068] [Q&A]
Hi ,
I'm pretty sure I have just confused myself, as I'm so far down this rabbit hole called RAG that I really don't know which way is up. So I thought I would ask the question .

Firstl my situation is that we have a large set of data for a clent which doesn't change.  Its at a certain date.  I would like to be able to ask question of the data and get answers.  Firstly from teh Open WebUI web site but secondly via API.  We are running Ollama and Open WebUI.

My question is that in Open WebUI there is the ability to upload documents into a knowledge base adn then create a model using the knowledge base.  Is this how we do RAG using Open WebUI?  Its just that all teh videos I have watched (quite a lot) have used n8n , or customer python.  They talk about spliting the files and manually inseting into vector databases.

It seem a bit too easy to use Open WebUI when all these peopel are releasing tutorial on how to do it.

Can someone please tell me if this is how to do RAG with Open Web Ui , cause it seems to work.

thanks
Greg

**Comment by ovizii** [comment]
That is definitely how you build a knowledge base, but I am also wondering about "ad-hoc" RA, meaning how to upload a file in chat and ask about it.
